go/libraries/doltcore/sqle/writer/prolly_write_session.go

```diff
@@ -37,10 +38,10 @@ type prollyWriteSession struct {
 	mut        *sync.RWMutex
 }
 
-var _ WriteSession = &prollyWriteSession{}
+var _ dsess.WriteSession = &prollyWriteSession{}
 
 // GetTableWriter implemented WriteSession.
-func (s *prollyWriteSession) GetTableWriter(ctx *sql.Context, table doltdb.TableName, db string, setter SessionRootSetter) (TableWriter, error) {
+func (s *prollyWriteSession) GetTableWriter(ctx *sql.Context, table doltdb.TableName, db string, setter dsess.SessionRootSetter) (dsess.TableWriter, error) {
```

Comment:
Same comment as above. This is really bad separation of concerns.

Looking around, it appears that this method is called in exactly one place where caching is critical: `(t *WritableDoltTable) getTableEditor`. That's where this caching logic should be applied.